server/api/endpoints/join-queries/table-join-endpoint.ts

```diff
@@ -0,0 +1,69 @@
+                // return all relevant info for the storing of the query
+                res.json({
+                    message: "Search results",
+                    result: resp,
+                    query: query,
+                    endpoint: 'table-join',
+                    category: `Joins movies_meta, movies_vote, and movie_credits`,
+                    search: req.body.search
+                })
+            }).catch(err => res.status(500).json({message: "Error", error: err}));
+            } else if (req.body.search === {selectMoviesMeta: false, selectMoviesVote: true, selectMoviesCredits: true})  {
+                const query = 'SELECT * FROM movies_vote FULL OUTER JOIN movies-credits ON movies_vote.vote_id = movies_credits.credit_id';
+                console.log('Query: ', query);
+                db.any(query, req.body.search).then( (resp) => {
+                // return all relevant info for the storing of the query
+                res.json({
+                    message: "Search results",
+                    result: resp,
+                    query: query,
+                    endpoint: 'table-join',
+                    category: `Joins movies_vote and movie_credits`,
+                    search: req.body.search
+                })
+            }).catch(err => res.status(500).json({message: "Error", error: err}));
+            } else if (req.body.search === {selectMoviesMeta: true, selectMoviesVote: false, selectMoviesCredits: true})  {
+                const query = 'SELECT * FROM movies_meta FULL OUTER JOIN movies_credits ON movies_meta.movie_id = movies_credits.credit_id';
+                console.log('Query: ', query);
+                db.any(query, req.body.search).then( (resp) => {
+                // return all relevant info for the storing of the query
+                res.json({
+                    message: "Search results",
+                    result: resp,
+                    query: query,
+                    endpoint: 'table-join',
+                    category: `Joins movies_meta and movie_credits`,
+                    search: req.body.search
+                })
+            }).catch(err => res.status(500).json({message: "Error", error: err}));
+            } else if (req.body.search === {selectMoviesMeta: true, selectMoviesVote: true, selectMoviesCredits: false})  {
+                const query = 'SELECT * FROM movies_vote FULL OUTER JOIN movies_meta ON movies_vote.vote_id = movies_meta.movie_id';
+                console.log('Query: ', query);
+                db.any(query, req.body.search).then( (resp) => {
+                // return all relevant info for the storing of the query
+                res.json({
+                    message: "Search results",
+                    result: resp,
+                    query: query,
+                    endpoint: 'table-join',
+                    category: `Joins movies_meta and movies_vote`,
+                    search: req.body.search
```

Comment:
There seems to be a lot of repeating code. Would it be possible to only have the if statements handle the query construction and the value of the category property and maybe put the rest down the line?

Reply:
I believe we trying to work that out now